bundles/org.openhab.binding.comfoair/src/main/java/org/openhab/binding/comfoair/internal/ComfoAirSerialConnector.java

```
@@ -0,0 +1,635 @@
+                                System.arraycopy(cleanedBlock, 0, _block, 0, _block.length);
+
+                                // validate calculated checksum against submitted
+                                // checksum
+                                if (calculateChecksum(_block) == checksum) {
+                                    if (logger.isTraceEnabled()) {
+                                        logger.trace("receive CMD: {} DATA: {}",
+                                                String.format("%02x", command.getReplyCmd()), dumpData(replyData));
+                                    }
+                                    send(ACK);
+
+                                    return replyData;
+                                }
+
+                                logger.debug("Unable to handle data. Checksum verification failed");
+                            } else {
+                                logger.debug("Unable to handle data. Data size not valid");
+                            }
+
+                            if (logger.isTraceEnabled()) {
+                                logger.trace("skip CMD: {} DATA: {}", String.format("%02x", command.getReplyCmd()),
+                                        dumpData(cleanedBlock));
+                            }
+                        }
+                    }
+                } catch (IOException e) {
+                    logger.debug("IO error: {}", e.getMessage());
+                }
+
+                try {
+                    Thread.sleep(1000);
+                    if (logger.isDebugEnabled()) {
+                        logger.debug("Retry cmd. Last call was not successful. Request: {} Response: {}",
+                                dumpData(requestBlock), (responseBlock.length > 0 ? dumpData(responseBlock) : "null"));
+                    }
+                } catch (InterruptedException e) {
+                    logger.warn("Transmission was interrupted: {}", e.getMessage());
```

If a thread receives an interrupt it generally means that it should terminate immediately. You shouldn't continue communication requests with it.

---

Not sure if interrupting the thread is enough or I also have to throw a RuntimeException or somthing. Pls check

---

@cpmeister I assume I also have to return a value here as well (preferably `return ComfoAirCommandType.Constants.EMPTY_INT_ARRAY`) or at least `break` the loop to quit communication requests (same for the other `InterruptedException`s)?

---

You could either break the loop or just throw an exception. Since any interrupt here wouldn't be caused by your own code I think the best route would be to throw a RuntimeException and let it propagate as high up the stack as possible, thus hopefully exiting as soon as possible.